packages/v2/contracts/ANTv2.sol

```diff
@@ -0,0 +1,159 @@
+pragma solidity 0.5.17;
+
+import './interfaces/IERC20.sol';
+import './libraries/SafeMath.sol';
+
+
+// Lightweight token modelled after UNI-LP: https://github.com/Uniswap/uniswap-v2-core/blob/v1.0.1/contracts/UniswapV2ERC20.sol
+// Adds:
+//   - A minting role
+//   - transferWithAuth()
+contract ANTv2 is IERC20 {
+    using SafeMath for uint256;
+
+    string private constant ERROR_NOT_MINTER = "ANTV2:NOT_MINTER";
+    string private constant ERROR_AUTH_EXPIRED = "ANTV2:AUTH_EXPIRED";
+    string private constant ERROR_AUTH_NOT_YET_VALID = "ANTV2:AUTH_NOT_YET_VALID";
+    string private constant ERROR_AUTH_ALREADY_USED = "ANTV2:AUTH_ALREADY_USED";
+    string private constant ERROR_INVALID_SIGNATURE = "ANTV2:INVALID_SIGNATURE";
+
+    string public constant name = "Aragon Network Token";
+    string public constant symbol = "ANT";
+    uint8 public constant decimals = 18;
+
+    address public minter;
+    uint256 public totalSupply;
+    mapping (address => uint256) public balanceOf;
+    mapping (address => mapping(address => uint256)) public allowance;
+
+    bytes32 public DOMAIN_SEPARATOR;
+
+    // bytes32 public constant PERMIT_TYPEHASH = keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public constant PERMIT_TYPEHASH = 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
+    mapping (address => uint256) public nonces;
+
+    // bytes32 public constant TRANSFER_WITH_AUTHORIZATION_TYPEHASH =
+    //     keccak256("TransferWithAuthorization(address from,address to,uint256 value,uint256 validAfter,uint256 validBefore,bytes32 nonce)");
+    bytes32 public constant TRANSFER_WITH_AUTHORIZATION_TYPEHASH = 0x7c7c6cdb67a18743f49ec6fa9b35f50d52ed05cbed4cc592e13b44501c1a2267;
+    mapping (address => mapping (bytes32 => bool)) public authorizationState;
```

Comment:
I didn't find a good reason to keep to the storage layout proposed in the EIP. Since this mapping is declared as `public`, it will generate the same interface as the `external` function laid out in the EIP.